### PR TITLE
Refactored touch mouse handler, added small keyboard updates

### PR DESCRIPTION
**Pull request by yoshisuga**
Added a dedicated touch mouse handler in Swift that supports mouse clicking by tapping, double-tap-and-hold to click and drag. Removed the old implementation.

Also made some keyboard updates: added the right-alt key, added tab key to alternate keyboard as well.

### CODE REVIEW

File: xcode/MAME4iOS/MAME4iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1352,6 +1368,7 @@
 					"$(PROJECT_DIR)",
 				);
 				OTHER_LDFLAGS = "";
+				PRODUCT_BUNDLE_IDENTIFIER = com.example.mame4iosx;
```
[the remark this answers is not in the document]
Comment (yoshisuga) [reply xcode/MAME4iOS/MAME4iOS.xcodeproj/project.pbxproj]: Done!

File: xcode/MAME4iOS/MAME4iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1390,6 +1407,7 @@
...
```
[the remark this answers is not in the document]
Comment (yoshisuga) [reply xcode/MAME4iOS/MAME4iOS.xcodeproj/project.pbxproj]: Done!

File: xcode/MAME4iOS/TouchMouse/EmulatorTouchMouse.swift [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+   var enabled = true
+   
+   let view: UIView
+   weak var delegate: EmulatorTouchMouseHandlerDelegate?
+   
+   private let positionChangeThreshold: CGFloat = 20.0
+   private let mouseHoldInterval: TimeInterval = 1.0
+   
+   private var pendingMouseEvents = [MouseClick]()
+   private var mouseEventPublisher: AnyPublisher<MouseClick, Never> {
```
[the remark this answers is not in the document]
Comment (yoshisuga) [reply xcode/MAME4iOS/TouchMouse/EmulatorTouchMouse.swift:53]: Haha I think it works ok in this case, nothing fancy.

File: iOS/EmulatorController.m [hunk 4]
```diff
@@ -2540,6 +2540,7 @@ - (void)changeUI { @autoreleasepool {
         [self.view makeToast:@"Touch Mouse Mode Enabled!" duration:2.0 position:CSToastPositionCenter
                        title:nil image:[UIImage systemImageNamed:@"cursorarrow.motionlines"] style:toastStyle completion:nil];
     }
+    [self.touchMouseHandler setEnabled:g_pref_touch_analog_enabled];
```
[the remark this answers is not in the document]
Comment (yoshisuga) [reply iOS/EmulatorController.m:2544]: This was in the file you sent, and the change is applied here, thanks.

File: xcode/MAME4iOS/TouchMouse/EmulatorTouchMouse.swift [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            secondaryTouch = nil
+         } else if touch == secondaryTouch?.touch {
+            if touch.tapCount > 0 {
+               mouseEventSubject.send(MouseClick(isRightClick: true, isPressed: true))
+               endHold()
+            }
+            secondaryTouch = nil
+         }
+      }
+      delegate?.handleMouseMove(x: 0, y: 0)
```
[the remark this answers is not in the document]
Comment (yoshisuga) [reply xcode/MAME4iOS/TouchMouse/EmulatorTouchMouse.swift:149]: The guard at the beginning takes care of this now, thanks.

**Review by ToddLa — APPROVED** [review]
Looks good, new delegate function is protected!

**Pull request merged**